### PR TITLE
Add pull to refresh functionality

### PR DESCRIPTION
**Pull request by Vanethos**
Add pull to refresh functionality on the occurrences list screen

### CODE REVIEW

File: lib/presentation/ui/home/home_bloc.dart [hunk 1]
```diff
@@ -4,15 +4,16 @@ import 'package:vost/domain/managers/occurrences_manager.dart';
...
-class HomeBloc extends BaseBloc {
+class HomeBloc extends BaseBloc with RefreshBlocMixin {
   static final recentsIndex = 0;
   static final followingIndex = 1;
 
   static final listIndex = 0;
   static final mapIndex = 1;
 
-  OccurrencesManager _countyManager;
+  OccurrencesManager _occurrenceManager;
```
Comment (miguelpruivo) [inline lib/presentation/ui/home/home_bloc.dart]: This can be `final`

File: lib/presentation/ui/home/home_page.dart [hunk 2]
```diff
@@ -185,32 +188,55 @@ class _MyHomePageState extends BaseState<HomePage> {
...
+
+class _RecentListWidgetState extends State<RecentListWidget> {
+  @override
+  void initState() {
+    super.initState();
+    widget.bloc.refreshErrorStream
+        .listen((_) => _refreshController.refreshFailed());
+  }
+
+  RefreshController _refreshController =
```
Comment (miguelpruivo) [inline lib/presentation/ui/home/home_page.dart]: This can be `final`.

File: lib/presentation/ui/home/home_page.dart [hunk 3]
```diff
@@ -185,32 +188,55 @@ class _MyHomePageState extends BaseState<HomePage> {
...
     return StreamBuilder<List<OccurrenceModel>>(
-        stream: bloc.occurrencesStream,
+        stream: widget.bloc.occurrencesStream,
         builder: (context, snapshot) {
           if (snapshot.connectionState == ConnectionState.waiting) {
             return Center(child: Text("A carregar"));
           }
 
+          _refreshController.refreshCompleted();
           if (snapshot.data != null) {
```
Comment (miguelpruivo) [inline lib/presentation/ui/home/home_page.dart]: This is from a previous commit, but I'd use here `snapshot.hasData` instead to make the null check more verbose.

**Review by miguelpruivo — CHANGES_REQUESTED** [review]

File: lib/presentation/ui/home/home_page.dart (hunk 3 above)
Comment (Vanethos) [reply lib/presentation/ui/home/home_page.dart]: yes! yoy're right, sorry it's the bad habbit

**Review by miguelpruivo — APPROVED** [review]
LGTM

**Pull request merged**